### PR TITLE
Godoc all the things

### PR DESCRIPTION
**Pull request by apg**
Produces godoc compatible documentation for the entire project. Builds on top of #7.

### CODE REVIEW

File: metrics.go [hunk 1]
```diff
@@ -2,43 +2,81 @@ package agentmon
...
 	Timer
 )
 
+// Measurement is a point in time value that is used to amend a metric.
 type Measurement struct {
-	Name      string
+	// Name is the metric to contribute to.
+	Name string
+
+	// Timestamp is the time at which the measurement was taken.
```
Comment (freeformz) [inline metrics.go]: `// Timestamp  at which the measurement was taken.`

File: metrics.go [hunk 2]
```diff
@@ -2,43 +2,81 @@ package agentmon
...
 	Timestamp time.Time
-	Type      MetricType
-	Value     float64
-	Sample    float32
-	Modifier  string
+
+	// Type is the type of the metric we wish to amend.
```
Comment (freeformz) [inline metrics.go]: `//Type of the metric we wish to ammend.`

Comment (apg) [inline metrics.go]: OK, fine.

File: metrics.go [hunk 3]
```diff
@@ -2,43 +2,81 @@ package agentmon
...
+	Type MetricType
+
+	// Value is the amount by which we will amend the metric.
+	// For Gauges, the amendment might be "replace".
+	Value float64
+
+	// Sample stores the sample rate that this measurement was taken at.
+	// For most applications, this should be set to 1.0
+	Sample float32
```
Comment (freeformz) [inline metrics.go]: maybe call this `SampleRate` instead and change the doc to:
```
SampleRate that this measurement was taken at.
```

File: metrics.go [hunk 4]
```diff
@@ -2,43 +2,81 @@ package agentmon
...
+
+	// Modifier allows gauges to be treated differently
+	//
+	// A value of "-" subtracts Value from the metric's previous value.
+	// A value of "+" adds Value to metric's previous value.  An empty
+	// value replaces the metric's value with Value.
+	Modifier string
```
Comment (freeformz) [inline metrics.go]: Maybe add `Any other values are ignored`?

File: prom/poller.go [hunk 5]
```diff
@@ -26,14 +26,28 @@ const (
 	promProto           = "io.prometheus.client.MetricFamily"
 )
 
+// Poller defined parameters for scraping a Prometheus endpoint, over HTTP.
 type Poller struct {
-	URL          *url.URL
-	Interval     time.Duration
+	// URL is the URL that the poller should scrape.
```
Comment (freeformz) [inline prom/poller.go]: `// URL that the poller should scrape.`

**Pull request merged**